core/store/src/config.rs

```diff
@@ -49,6 +47,23 @@ pub struct StoreConfig {
     pub trie_cache_capacities: Vec<(ShardUId, usize)>,
 }
 
+/// Mode in which to open the storage.
+#[derive(Clone, Copy)]
+pub enum Mode {
+    ReadOnly,
+    ReadWrite,
+}
+
+impl Mode {
+    pub fn read_write(rw: bool) -> Self {
```

Comment:
I often find that using `bool` in such ways makes code reading more challenging.  Instead should we consider having two constructors instead.  Something like the following.

```
fn read_write() -> Self {Self::ReadWrite}
fn read_only() -> Self {Self::ReadOnly}
```

Reply:
It seems like this function is used in only a single place where we parse cmdline arguments.  I would rather retain the existing style where we keep the `bool` related ambiguity in that file rather than introducing a public constructor that might get used in other places in future.